src/app/content/components/SkipToContent.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { AppState } from '../../types';
+import * as select from '../selectors';
+
+const HiddenLink = styled.a`
+    /* Hide the link when it is not focused */
+    clip: rect(1px,1px,1px,1px);
+    margin: 0;
+    position: absolute;
+    left: 0;
+    top: 0;
+    height: 1px;
+    width: 1px;
+    overflow: hidden;
+
+    text-decoration: none;
+
+    /* Show the link when it is focused */
+    :focus {
+        clip: auto;
+        height: auto;
+        width: auto;
+        z-index: 20;
+    }
+`;
+
+interface PropTypes {
+  targetId: string;
+  loading: boolean;
+}
+
+class SkipToContent extends Component<PropTypes, never> {
+
+  public render() {
+    const { targetId } = this.props;
+    return !this.isLoading() && <HiddenLink href={`#${targetId}`} tabIndex={1}>Skip to Content</HiddenLink>;
```

Comment:
if we scroll with js instead, can we pass a ref to the target element into `SkipToContent` instead of relying on the id and hash?

Reply:
That would be cleaner... I can try although I'm not sure how to pass a ref to the target element since it is constructed in a child component(my React-fu is fuzzy).

Also, I think we do need to support the href & `id="..."` as fallback if JS did not load

Reply:
do we support not js not being loaded? that will affect many other things

Reply:
i think the reacty way to do it would be to have a `<SkipToContentWrapper>` component that rendered the link before its children, and a `<SkipToContentTarget>` HOC that wrapped your actual target, then use [context](https://reactjs.org/docs/context.html) to communicate between the two and use [find dom node](https://reactjs.org/docs/react-dom.html#finddomnode) on `SkipToContentTarget`'s child to get the dom reference to scroll too.

i know it sounds more complex, but it'll enable using the SkipToContent pattern on more pages with less id wrangling

Reply:
One concern with that is that the SkipToContent link needs to appear at the beginning of the DOM (before the navigation, the ToC, etc) while the actual content element lives after the navigation, the ToC. Would the pattern you mentioned still work? If so, it seems like the `<SkipToContentWrapper>` would need to surround the entire app, right?


RE Not having JS enabled: What else would break? (I am assuming that only non-logged-in users would not have JS enabled)

Reply:
if we do want it to work without js, `<SkipToContentTarget>` could inject a anchor with a randomly generated id and communicate that back to `<SkipToContentWrapper>` via context on `componentWillMount`. all that will get baked into the pre-rendered html so then the link will work without js, but i still think we should have a js handler so we don't get the url change when we can avoid it

Reply:
I added a JS handler but was wondering if it is worth adding the `react-scroll` package or just using the part of it that we need (seen in b277764) befoe I try writing tests.